trunner/tools/gdb.py

```diff
@@ -94,30 +94,32 @@ class OpenocdError(ProcessError):
 class OpenocdGdbServer:
     """Handler for OpenocdGdbServer process"""
 
-    def __init__(self, interface: str, target: str):
+    def __init__(self, interface: str, target: str, extra_args: Optional[list[str]] = None):
         self.proc = None
         self.target = target
         self.interface = interface
         self.output = ""
         self.logfile = io.StringIO()
+        self.extra_args = extra_args
```

Comment:
I am not sure `if self.extra_args` it's working, but maybe we can extend arguments even with empty list (then if won't be needed)

```suggestion
        self.extra_args = []
```

Reply:
It's working, `None` and `[]` (empty list) evaluates to False in python

Reply:
There was a mistake in my comment, now it's corrected.